### PR TITLE
configure sonarcloud

### PR DESCRIPTION
**Pull request by tyrasd**
### Description
sets up sonarcloud static code analysis: https://sonarcloud.io/dashboard?id=org.heigit.ohsome%3Aoshdb

### New or changed dependencies
- update parent to version 2.9

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -61,6 +61,11 @@ pipeline {
     stage ('Reports and Statistics') {
       steps {
         script {
+          withSonarQubeEnv('sonarcloud GIScience/ohsome') {
+            // START CUSTOM oshdb
+            sh "mvn sonar:sonar -Dsonar.branch.name=${env.BRANCH_NAME} -Dsonar.projectName=OSHDB"
+            // END CUSTOM oshdb
+          }
```
Comment (joker234) [inline Jenkinsfile:68]:
```suggestion
          // START CUSTOM oshdb
          withSonarQubeEnv('sonarcloud GIScience/ohsome') {
            sh "mvn sonar:sonar -Dsonar.branch.name=${env.BRANCH_NAME} -Dsonar.projectName=OSHDB"
          }
          // END CUSTOM oshdb
```

File: pom.xml [hunk 2]
```diff
@@ -46,6 +52,8 @@
     <roaringbitmap.version>0.6.59</roaringbitmap.version>
     <rxjava2.version>2.1.9</rxjava2.version>
     <slf4j.version>1.7.25</slf4j.version>
+    <sonar.moduleKey>${project.artifactId}</sonar.moduleKey>
+    <sonar.projectKey>org.heigit.ohsome:oshdb</sonar.projectKey>
```
Comment (joker234) [inline pom.xml:56]: Does something like this work here?
```suggestion
    <sonar.projectKey>${project.groupId}:oshdb</sonar.projectKey>
```

**Review by joker234 — CHANGES_REQUESTED** [review]

**Comment by joker234** [comment]
Maybe additionally: Do we want to add the parent change to the Changelog?

File: pom.xml (hunk 2 above)
Comment (tyrasd) [reply pom.xml:56]: let's try

**Comment by tyrasd** [comment]
> Maybe additionally: Do we want to add the parent change to the Changelog?

IMHO such changes are not really something users need to know (and would search in a changelog document), except if it is a big/significant change which affects developers directly (like the change of the group id). but since we had a changelog entry for a parent upgrade already, I just generified that one a bit. :innocent:

**Review by joker234 — APPROVED** [review]

**Pull request merged**